html/feed/feed.js

```diff
@@ -1,5 +1,369 @@
+
+    modal.show ();
+  }); 
+
+  btnGroup.appendChild (btnEdit);
+}
+
+// likeBtn eventListener 
+
+btnLike.addEventListener ("click", (e)=>{
+  reactToPosts (baseUrl, element.id, token, spanlike)
+})
+
+btnGroup.appendChild (btnComment); 
+btnGroup.appendChild (btnLike); 
+
+cardBody.appendChild (btnGroup); 
+
+colContent.appendChild (cardBody); 
+
+row.appendChild (colImage); 
+row.appendChild (colContent); 
+
+card.appendChild (row); 
+
+// commentSection 
+
+const commentCollapse = document.createElement ('div'); 
+commentCollapse.className = 'collapse mt-3'; 
+commentCollapse.id = `commentCollapse${element.id}`; 
+
+// create the form 
+
+const commentForm = document.createElement ('form'); 
+
+const commentInput = document.createElement ('input'); 
+commentInput.className = 'form-control'; 
+commentInput.type = 'text'; 
+commentInput.placeholder = 'Write a reply...'; 
+commentForm.appendChild (commentInput); // append to form insted 
+
+// add a hidden submit btn- allows users to press enter to submit. 
+
+const hiddenSubmitBtn = document.createElement ('button'); 
+hiddenSubmitBtn.style.display = 'none'; 
+commentForm.appendChild (hiddenSubmitBtn); 
+
+commentForm.addEventListener ("submit", (e)=> {
+  e.preventDefault ()
+    const commentText = commentInput.value; 
+    if (commentText){
+      postComment (baseUrl, element.id, commentText, token, spancomment, element); 
+      commentInput.value = ''; 
+
+    }
+
+})
+
+commentCollapse.appendChild (commentForm); 
+
+//modify btnComment to controll the collapse of comment section 
+
+btnComment.setAttribute ('data-bs-toggle','collapse'); 
+btnComment.setAttribute ('data-bs-target','#commentCollapse${element.id}'); 
+btnComment.setAttribute ('aria-expanded','false'); 
+btnComment.setAttribute ('aria-controls','commentCollapse${element.id}'); 
+    
+cardBody.appendChild (commentCollapse); 
+
+//commentSectionPanel to store comments 
+
+const commentsPanel = document.createElement ('div'); 
+commentsPanel.className= 'comments-panel border-top mt-2'; 
+commentsPanel.style.display = 'none'; // initially hidden panel 
+
+element.comments.forEach (comment => {
+  const commentDiv = document.createElement ('div'); 
+  commentDiv.className = 'comment py-2 d-flex gap-2 align-items-center'; // = flex container
+
+  const commentUser = document.createElement ('strong'); 
+  commentUser.textContent = comment.author.name + ": "; 
+  commentDiv.appendChild (commentUser); 
+
+  const commentText = document.createElement ('span'); 
+  commentText.textContent = comment.body; 
+  commentDiv.appendChild (commentText); 
+
+  commentsPanel.appendChild (commentDiv); 
+}); 
+
+  cardBody.appendChild (commentsPanel); 
+
+  btnComment.addEventListener ('click', function(){
+    if (commentsPanel.style.display === 'none'){
+      commentsPanel.style.display = 'block'; 
+    } else { 
+      commentsPanel.style.display = 'none'; 
+    } 
+
+  container.appendChild (card); 
```

Comment:
you are appending the html to main inside a eventlistner for the comment button. please check your scoping when writing functions. the container which is the main tagg in this case, appends this card component when the function runs. if it only appends the the card when the comment button is pressed which doesnt exist yet because main has not yet appended the html, then it cant render :P move this out of the eventlistener please.

Reply:
Thank you. I see the error. I moved the append to be in the function.

Reply:
its still not in the correct part, the card is defined in the forEach part of the function. meaning forEach post it creats a card element, we then need to append the Card element each time a card has been made. now you are appending one card that doesnt exist to the container element outside the forEach function. move it inside the forEach function and it should work.
![image](https://github.com/AnaCascante/2023-10-15_JS2_CA_AnaC_JeanC_Repo/assets/89041011/4eba9f18-0940-4a31-88a2-3e6b2c8063d7)

Reply:
fixed